service/controllers/contactsController.js

```diff
@@ -1,63 +1,87 @@
 const { Contact } = require("../Schemas/contactSchema");
 
-const listContacts = async () => {
+const listContacts = async (owner) => {
   try {
-    const contacts = await Contact.find();
+    const contacts = await Contact.find({ owner });
     return contacts;
   } catch (err) {
     throw Error(err);
   }
 };
 
-const getContactById = async (contactId) => {
+const getContactById = async (contactId, userId) => {
   try {
     const contactToFind = await Contact.findOne({ _id: contactId });
+
+    if (contactToFind && userId.toString() !== contactToFind.owner.toString()) {
+      throw Error("Not found in your contacts!");
+    }
     return contactToFind;
   } catch (err) {
     throw Error(err);
   }
 };
 
-const removeContact = async (contactId) => {
+const removeContact = async (contactId, userId) => {
   try {
     const contactToDelete = await Contact.findByIdAndDelete({ _id: contactId });
```

Comment:
```suggestion
    const contactToDelete = await Contact.findByIdAndDelete({ _id: contactId });
```
Powinnaś jeszcze sprawdzić czy kontakt należy do owner

Reply:
Przeciez jest dalej sprawdzenie owner?